### DISCUSSION TITLE
Best model found on immediately continuing training

### DISCUSSION THREAD

**Discussion by skol101** [coqui-ai/TTS#967] [General Q&A]
I observed this behaviour when stopping and continuing training on a VITS VCTK model. 

Previous best model was found around step 261k

> Restoring best loss from best_model_261377.pth.tar ...
 > Starting with loaded last best loss 16.772456209161387.

I stopped training when the last saved checkpoint check checkpoint_340k. 
I restart training and here's what I observe:
`/home/sk/anaconda3/envs/tts_one/lib/python3.8/site-packages/torch/optim/lr_scheduler.py:122: UserWarning: Seems like `optimizer.step()` has been overridden after learning rate scheduler initialization. Please, make sure to call `optimizer.step()` before `lr_scheduler.step()`. See more details at https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate
  warnings.warn("Seems like `optimizer.step()` has been overridden after learning rate scheduler "
/home/sk/anaconda3/envs/tts_one/lib/python3.8/site-packages/torch/optim/lr_scheduler.py:122: UserWarning: Seems like `optimizer.step()` has been overridden after learning rate scheduler initialization. Please, make sure to call `optimizer.step()` before `lr_scheduler.step()`. See more details at https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate
  warnings.warn("Seems like `optimizer.step()` has been overridden after learning rate scheduler "

 > DataLoader initialization
 | > Use phonemes: True
   | > phoneme language: en-us
 | > Number of instances : 191
 | > Computing phonemes ...
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████| 191/191 [00:00<00:00, 9469.63it/s]
 | > Max length sequence: 263945.0
 | > Min length sequence: 17221.0
 | > Avg length sequence: 57104.17277486911
 | > Num. instances discarded by max-min (max=1500000, min=32768) seq limits: 23
 | > Batch group size: 0.

 > EVALUATION 

 | > Synthesizing test sentences.

  --> EVAL PERFORMANCE
     | > avg_loader_time: 0.00820 (+0.00000)
     | > avg_loss_gen: 2.29878 (+0.00000)
     | > avg_loss_kl: 2.18174 (+0.00000)
     | > avg_loss_feat: 6.04929 (+0.00000)
     | > avg_loss_mel: 18.55187 (+0.00000)
     | > avg_loss_duration: 1.69043 (+0.00000)
     | > avg_loss_0: 30.77211 (+0.00000)
     | > avg_loss_disc: 2.41310 (+0.00000)
     | > avg_loss_1: 2.41310 (+0.00000)

 > BEST MODEL : vits_vctk_all-November-22-2021_04+03PM-0000000/best_model_340243.pth.tar
`

**Comment by Ca-ressemble-a-du-fake** [comment]
@skol101 did you find an explanation ? I am observing the same behaviour.

**Reply by e0xextazy** [reply]
Most likely, as written here [#1039](https://github.com/coqui-ai/TTS/issues/1039#issuecomment-1036213275), LR Resets when you resume training

**Reply by Ca-ressemble-a-du-fake** [reply]
Great, there is an explanation for all :)!
